src/main/java/com/bobocode/petros/scaner/AnnotationDependencyConfigurationScanner.java

```diff
@@ -68,24 +64,41 @@ public AnnotationDependencyConfigurationScanner() {
+    private Map<String, List<Method>> dependencyMethodsByConfigClasses(List<Class<?>> configClasses) {
         return configClasses.stream()
-                .flatMap(cl -> Arrays.stream(cl.getDeclaredMethods()))
+                .collect(Collectors.toMap(Class::getName, this::getDependencyMethods));
+    }
+
+    private List<Method> getDependencyMethods(Class<?> clazz) {
+        return Arrays.stream(clazz.getDeclaredMethods())
                 .filter(method -> method.isAnnotationPresent(Dependency.class))
                 .collect(Collectors.toList());
     }
 
-    private Map<String, List<DependencyDefinition>> groupMethodsDependencyDefinitions(List<Method> methods) {
-        for (Method method : methods) {
-            var returnTypeDependencyDefinition = new DependencyDefinition();
-            returnTypeDependencyDefinition.setName(getDependencyNameFrom(method));
-            returnTypeDependencyDefinition.setConfigClassDependency(true);
-            var returnTypeQualifiedName = method.getReturnType().getName();
-            returnTypeDependencyDefinition.setQualifiedName(returnTypeQualifiedName);
-            setInjectedDependenciesDefinitionsFromParameters(returnTypeDependencyDefinition, method.getParameters());
-            configurationsMap.computeIfAbsent(returnTypeQualifiedName, k -> new ArrayList<>()).add(returnTypeDependencyDefinition);
-        }
-        return configurationsMap;
+    private Map<String, List<DependencyDefinition>> groupMethodsDependencyDefinitions(Map<String, List<Method>> methods) {
+        return methods.entrySet().stream()
+                .map(this::dependencyDefinitionsFrom)
+                .flatMap(Collection::stream)
+                .collect(Collectors.groupingBy(
+                        DependencyDefinition::getQualifiedName,
+                        Collectors.mapping(identity(), Collectors.toList())
+                ));
+    }
+
+    private List<DependencyDefinition> dependencyDefinitionsFrom(Map.Entry<String, List<Method>> entry) {
+        return entry.getValue().stream()
+                .map(method -> {
+                    var returnTypeDependencyDefinition = new DependencyDefinition();
+                    returnTypeDependencyDefinition.setName(getDependencyNameFrom(method));
+                    returnTypeDependencyDefinition.setConfigClassDependency(true);
+                    returnTypeDependencyDefinition.setInjectedDependencyMethodName(method.getName());
+                    var returnTypeQualifiedName = method.getReturnType().getName();
+                    returnTypeDependencyDefinition.setQualifiedName(returnTypeQualifiedName);
+                    setInjectedDependenciesDefinitionsFromParameters(returnTypeDependencyDefinition, method.getParameters());
+                    returnTypeDependencyDefinition.setConfigClassQualifiedName(entry.getKey());
+                    return returnTypeDependencyDefinition;
+                })
```

Comment:
On my point of view this piece of code must be in method helper.

Reply:
Agree with Markian, 9 lines in lambda it's too much. Could you please, extract it into separate method

Reply:
Sure, makes sence